### PR TITLE
MDEXP-227: MappingProfile: Display contributor name fields 

### PR DESCRIPTION
**Pull request by KVupp**
https://issues.folio.org/browse/MDEXP-227

## Purpose
Display Contributor name fields with mapping for primary and non-primary fields

## Approach
1. Assign a json path for contributor name types
2. Added a test (currently only for instances) which will create a mapping profile and then a marc record based on the transformation fields so that the paths we specify there are validated
3. The item transformation paths all have "item" as a path but in our json we are defining it as items

As part of that test, discovered there were several incorrect paths, this PR has corrections to those as well.

## TODO
- We still need to write tests for ITEMS and HOLDINGS transformations, which we can handle in later stories

## Learning
- used https://jsonpath.com/ to validate the path specified

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/transformationfields/TransformationFieldsConfig.java [hunk 1]
```diff
@@ -24,21 +24,21 @@
...
 
   //Instance specific fields
-  ALTERNATIVE_TITLES("alternativeTitleTypeId", "$.instance[*].alternativeTitles[?(@.alternativeTitleTypeId=={id})]", ALTERNATIVE_TITLE_TYPES),
-  CONTRIBUTOR_NAME_TYPE("contributorNameTypeId", "$.instance[*].contributorNameTypeId[?(@.contributorNameTypeId=={id} && ?(!(@.primary) || @.primary == false))]", CONTRIBUTOR_NAME_TYPES),
-  CONTRIBUTOR_NAME_TYPE_PRIMARY("contributorNameTypeId.primary", "$.instance[*].contributorNameTypeId[?(@.contributorNameTypeId=={id} && ?((@.primary) && @.primary == true))]", CONTRIBUTOR_NAME_TYPES),
+  ALTERNATIVE_TITLES("alternativeTitleTypeName", "$.instance.alternativeTitles[?(@.alternativeTitleTypeId=='{id}')].name", ALTERNATIVE_TITLE_TYPES),
+  CONTRIBUTOR_NAME("contributorName", "$.instance.contributors[?(@.contributorNameTypeId=='{id}' && (!(@.primary) || @.primary == false))].name", CONTRIBUTOR_NAME_TYPES),
+  CONTRIBUTOR_NAME_PRIMARY("contributorName.primary", "$.instance.contributors[?(@.contributorNameTypeId=='{id}' && ((@.primary) && @.primary == true))].name", CONTRIBUTOR_NAME_TYPES),
```
Comment (illia-borysenko) [inline src/main/java/org/folio/service/transformationfields/TransformationFieldsConfig.java:39]: I found that true/false statements in jsonpath should be in single quotes '' to evaluate path correctly

Comment (illia-borysenko) [reply src/main/java/org/folio/service/transformationfields/TransformationFieldsConfig.java:39]: And we also found that quotes for {id} isn't required

Comment (KVupp) [reply src/main/java/org/folio/service/transformationfields/TransformationFieldsConfig.java:39]: Quotes are needed Illia
Screen Shot 2020-08-20 at 8 44 36 AM
Screen Shot 2020-08-20 at 8 44 45 AM

Comment (KVupp) [reply src/main/java/org/folio/service/transformationfields/TransformationFieldsConfig.java:39]: Check here https://github.com/json-path/JsonPath#filter-operators
> String literals must be enclosed by single or double quotes ([?(@.color == 'blue')] or [?(@.color == "blue")]).

UUID is a String in our schema

Comment (illia-borysenko) [reply src/main/java/org/folio/service/transformationfields/TransformationFieldsConfig.java:39]: Yeah, you are right, sorry for wrong comment

**Review by Igor-Gorchakov — APPROVED** [review]

**Review by illia-borysenko — APPROVED** [review]

**Pull request merged**